File: TheaterSuggestions/CSharp/SeatsSuggestions.Domain/SeatingOptionSuggested.cs
using System.Collections.Generic;

namespace SeatsSuggestions.Domain
{
    public class SeatingOptionSuggested
    {
        public PricingCategory PricingCategory { get; }
        public List<Seat> Seats { get; } = new List<Seat>();
        public PartyRequested PartyRequested { get; }

        public SeatingOptionSuggested(SuggestionRequest suggestionRequest)
        {
            PartyRequested = suggestionRequest.PartyRequested;
            PricingCategory = suggestionRequest.PricingCategory;
        }

        public bool MatchExpectation()
        {
            return Seats.Count == PartyRequested.PartySize;
        }

        public void AddSeats(AdjacentSeats adjacentSeats)
        {
            Seats.AddRange(adjacentSeats);
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestions/src/main/java/com/baasie/SeatsSuggestions/SeatingOptionSuggested.java
package com.baasie.SeatsSuggestions;

import java.util.ArrayList;
import java.util.List;

public class SeatingOptionSuggested {

    private final PricingCategory pricingCategory;
    private final List<Seat> seats = new ArrayList<>();
    private final int partyRequested;

    public SeatingOptionSuggested(SuggestionRequest suggestionRequest) {
        this.pricingCategory = suggestionRequest.pricingCategory();
        this.partyRequested = suggestionRequest.partyRequested();
    }

    public void addSeat(Seat seat) {
        seats.add(seat);
    }

    public boolean matchExpectation() {
        return seats.size() == partyRequested;
    }

    public List<Seat> seats() {
        return seats;
    }
}
